module-bsp/board/rt1051/bsp/eink/BspEink.cpp

```diff
@@ -0,0 +1,478 @@
+        spi_config.dataOutConfig                 = kLpspiDataOutRetained;
+
+        LPSPI_MasterInit(lpspiBase, &spi_config, lpspiGetFreq());
+
+        /* FSL LPSPI doesn't support configuring autopcs feature */
+        lpspiBase->CFGR1 |= LPSPI_CFGR1_AUTOPCS(0);
+
+        dmamux = DriverDMAMux::Create(static_cast<DMAMuxInstances>(BoardDefinitions::EINK_DMAMUX), {});
+        dma    = DriverDMA::Create(static_cast<DMAInstances>(BoardDefinitions::EINK_DMA), {});
+
+        txDMAHandle = dma->CreateHandle(static_cast<std::uint32_t>(BoardDefinitions::EINK_TX_DMA_CHANNEL));
+        rxDMAHandle = dma->CreateHandle(static_cast<std::uint32_t>(BoardDefinitions::EINK_RX_DMA_CHANNEL));
+
+        dmamux->Enable(static_cast<std::uint32_t>(BoardDefinitions::EINK_TX_DMA_CHANNEL),
+                       BSP_EINK_LPSPI_DMA_TX_PERI_SEL); // TODO: M.P fix BSP_EINK_LPSPI_DMA_TX_PERI_SEL
+        dmamux->Enable(static_cast<std::uint32_t>(BoardDefinitions::EINK_RX_DMA_CHANNEL),
+                       BSP_EINK_LPSPI_DMA_RX_PERI_SEL); // TODO: M.P fix BSP_EINK_LPSPI_DMA_RX_PERI_SEL
+
+        lpspiEdmaDriverState.edmaTxDataToTxRegHandle = reinterpret_cast<edma_handle_t *>(txDMAHandle->GetHandle());
+        lpspiEdmaDriverState.edmaRxRegToRxDataHandle = reinterpret_cast<edma_handle_t *>(rxDMAHandle->GetHandle());
+
+        LPSPI_SetMasterSlaveMode(lpspiBase, kLPSPI_Master);
+        LPSPI_MasterTransferCreateHandleEDMA(lpspiBase,
+                                             lpspi->handle,
+                                             lpspiMasterEdmaCallback,
+                                             nullptr,
+                                             lpspi->edmaRxRegToRxDataHandle,
+                                             lpspi->edmaTxDataToTxRegHandle);
+
+        isInitialized = true;
+        return kStatus_Success;
+    }
+
+    void deinit()
+    {
+        if (!isInitialized) {
+            return;
+        }
+
+        LPSPI_Enable(lpspiBase, false);
+
+        if (busySemaphore != nullptr) {
+            vSemaphoreDelete(busySemaphore);
+            busySemaphore = nullptr;
+        }
+
+        if (transferCompleteQueue != nullptr) {
+            vQueueDelete(transferCompleteQueue);
+            transferCompleteQueue = nullptr;
+        }
+
+        dmamux->Disable(static_cast<std::uint32_t>(BoardDefinitions::EINK_TX_DMA_CHANNEL));
+        dmamux->Disable(static_cast<std::uint32_t>(BoardDefinitions::EINK_RX_DMA_CHANNEL));
+        dmamux.reset();
+        dma.reset();
+
+        gpio->DisableInterrupt(1 << static_cast<std::uint32_t>(BoardDefinitions::EINK_BUSY_PIN));
+        gpio->ClearPortInterrupts(1 << static_cast<std::uint32_t>(BoardDefinitions::EINK_BUSY_PIN));
+        gpio.reset();
+
+        isInitialized = false;
+    }
+
+    void logicPowerOn()
+    {
+        bsp::eink::eink_gpio_power_on();
+    }
+
+    void logicPowerOff()
+    {
+        bsp::eink::eink_gpio_power_off();
```

Comment:
`bsp::eink` namespace is not needed here

Reply:
Done